Frontend/ui-customer/src/Flow.purs

```diff
@@ -538,138 +556,149 @@ currentFlowStatus = do
+                  else if ( err.code == 429 && codeMessage == "HITS_LIMIT_EXCEED") then do
+                      pure $ toast (getString STR.TOO_MANY_LOGIN_ATTEMPTS_PLEASE_TRY_AGAIN_LATER)
+                      modifyScreenState $ EnterMobileNumberScreenType (\enterMobileNumberScreen → enterMobileNumberScreen {props {enterOTP = false, wrongOTP = false}, data{otp=""}})
+                  else do
+                      pure $ toast (getString STR.SOMETHING_WENT_WRONG_PLEASE_TRY_AGAIN)
+                      modifyScreenState $ EnterMobileNumberScreenType (\enterMobileNumberScreen → enterMobileNumberScreen {props {enterOTP = false,wrongOTP = false}, data{otp=""}})
+                  enterMobileNumberScreenFlow
+      GoToOTP state -> do
+        when ((getValueToLocalStore MOBILE_NUMBER) /= state.data.mobileNumber)
+          $ do
+              deleteValueFromLocalStore SUGGESTIONS_MAP
+              deleteValueFromLocalStore RECENT_SEARCHES
+        void $ pure $ setValueInWindow (show MOBILE_NUMBER) state.data.mobileNumber
+        setValueToLocalStore COUNTRY_CODE (state.data.countryObj.countryCode)
+        void $ liftFlowBT $ setCleverTapUserData "Phone" (state.data.countryObj.countryCode <> (getValueToLocalStore MOBILE_NUMBER))
+        (TriggerOTPResp triggerOtpResp) <- Remote.triggerOTPBT (Remote.makeTriggerOTPReq state.data.mobileNumber state.data.countryObj.countryCode (show state.data.otpChannel))
+        void $ pure $ toast (getString if state.data.otpChannel == SMS then STR.SENT_OTP_VIA_SMS else STR.SENT_OTP_VIA_WHATSAPP)
+        modifyScreenState $ EnterMobileNumberScreenType (\enterMobileNumberScreen → enterMobileNumberScreen { data { tokenId = triggerOtpResp.authId, attempts = triggerOtpResp.attempts }, props { enterOTP = true, resendEnable = false } })
+        modifyScreenState $ HomeScreenStateType (\homeScreen → homeScreen { data { settingSideBar { number = state.data.mobileNumber } } })
+        enterMobileNumberScreenFlow
+      ResendOTP state -> do
+        (ResendOTPResp resendResp) <- Remote.resendOTPBT state.data.tokenId
+        modifyScreenState $ EnterMobileNumberScreenType (\enterMobileNumberScreen → enterMobileNumberScreen { data { tokenId = resendResp.authId, attempts = resendResp.attempts } })
+        enterMobileNumberScreenFlow
+      GoBack state -> do
+        modifyScreenState $ EnterMobileNumberScreenType (\enterMobileNumberScreen → enterMobileNumberScreen { data { timer = 30 }, props { enterOTP = false, resendEnable = false } })
+        enterMobileNumberScreenFlow
+      GoToWelcomeScreen state -> welcomeScreenFlow
 
 welcomeScreenFlow :: FlowBT String Unit
 welcomeScreenFlow = do
-  hideLoaderFlow
-  flow <- UI.welcomeScreen
-  case flow of
-    GoToMobileNumberScreen -> enterMobileNumberScreenFlow
+  config <- CP.getAppConfigFlowBT CP.appConfig
```

Comment:
We can create common function for this, instead of writing it everywhere

kind of

checkConfigAndCallFlow :: FlowBT String Unit -> FlowBT String Unit
checkConfigAndCallFlow flow =
config <- CP.getAppConfigFlowBT CP.appConfig
  if config.reactNativeFlowEnabled 
    then pure unit 
  else flow